fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1907,6 +1979,15 @@ ACTOR static Future<Void> transactionStarter(MasterProxyInterface proxy,
 
 			if (req.priority < TransactionPriority::DEFAULT &&
 			    !batchRateInfo.canStart(transactionsStarted[0] + transactionsStarted[1], tc)) {
+				if (SERVER_KNOBS->BATCH_TXN_FIX_TURN_ON) {
+					req.reply.sendError(batch_transaction_throttled());
```

Comment:
I don't believe we want to return an error here unconditionally. Batch transactions should behave like normal transactions and delay if the cluster is still in the process of acquiring more batch capacity. If the rate is 0, then we would return an error because there's no possibility that we will be able to proceed.